### PR TITLE
test(app): cover Library empty and error states

### PR DESCRIPTION
**Pull request by VincentShipsIt**
## Summary

- distinguish empty Library results from query failures across shared media, captions, and voices
- add recoverable error alerts with retry actions while preserving populated and canonical empty states
- add focused React Testing Library and hook coverage for empty, populated, failure, and recovery paths

## Verification

- `bunx biome check <12 touched files>` — passed
- `bun run check:design-system` — passed
- `git diff --cached --check` — passed
- staged filename and credential-pattern scan — passed
- local Secretlint attempted but unavailable because its configured rule packages are not installed in this dependency-light worktree; PR CI runs Secretlint and Gitleaks
- local Vitest/typecheck not run under the repository workstation policy; PR CI is the required broad verification gate

## Scope

- Parent: #1594
- No API contract changes

Closes #1945

### CODE REVIEW

**Comment by VincentShipsIt** [comment]
Fleet Review: **PASS** (Opus verifier fallback after Fable provider refusal).

- Blocking findings: none
- Linked issue: #1945 acceptance criteria satisfied
- Follow-up verification: exact-head source confirms Button/ButtonVariant and fireEvent imports exist; the required return type has no separate object-construction consumers.

File: apps/app/app/(protected)/[orgSlug]/[brandSlug]/library/captions/captions-list.test.tsx [hunk 1]
```diff
@@ -42,6 +43,13 @@ vi.mock('@services/core/notifications.service', () => ({
   },
 }));
 
+vi.mock('@services/core/logger.service', () => ({
+  logger: {
+    error: loggerErrorMock,
+    info: vi.fn(),
+  },
+}));
```
[the remark this answers is not in the document]
Comment (VincentShipsIt) [reply apps/app/app/(protected)/[orgSlug]/[brandSlug]/library/captions/captions-list.test.tsx:57]: Verified against exact head bcdd4a35a565fa47d6fc42e29187341ff94d5e4c: `findAllMock` is not evaluated by the hoisted factory. The factory only creates a nested async closure; `findAllMock` is dereferenced when that closure runs during the test, after top-level initialization. `loggerErrorMock`, which is dereferenced directly during factory execution, is already created with `vi.hoisted`. Exact-head app tests and Typecheck are successful, so no TDZ occurs. No code change is warranted.
